From pavel@ucw.cz  Fri Dec 18 10:11:52 2009
From: Pavel Machek <pavel@ucw.cz>
Date: Sat, 21 Nov 2009 09:16:47 +0100
Subject: Staging: dream: fix memory leak in camera error path
To: Arve Hj?nnev?g <arve@android.com>, Brian Swetland <swetland@google.com>, San Mehat <san@google.com>, Daniel Walker <dwalker@fifo99.com>, Iliyan Malchev <malchev@google.com>, Greg KH <greg@kroah.com>
Message-ID: <20091121081647.GC2597@elf.ucw.cz>
Content-Disposition: inline



cppcheck found that ctrl_pmsm is leaked if the open operation fails.

Signed-off-by: Eric Sesterhenn <eric.sesterhenn@lsexperts.de>
Signed-off-by: Pavel Machek <pavel@ucw.cz>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/dream/camera/msm_camera.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/staging/dream/camera/msm_camera.c
+++ b/drivers/staging/dream/camera/msm_camera.c
@@ -1885,8 +1885,10 @@ static int msm_open_control(struct inode
 		return -ENOMEM;
 
 	rc = msm_open_common(inode, filep, 0);
-	if (rc < 0)
+	if (rc < 0) {
+		kfree(ctrl_pmsm);
 		return rc;
+	}
 
 	ctrl_pmsm->pmsm = filep->private_data;
 	filep->private_data = ctrl_pmsm;
